trieste/acquisition/function.py

```diff
@@ -130,6 +133,110 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
     return (eta - mean) * normal.cdf(eta) + variance * normal.prob(eta)
 
 
+class MaxValueEntropySearch(SingleModelAcquisitionBuilder):
+    """
+    Builder for the max-value entropy search acquisition function (for function minimisation)
+    """
+
+    def __init__(self,search_space: SearchSpace, num_samples: int = 10, grid_size: int = 5000):
+        """
+        :param search_space: The search space over which the acquisiiton function is defined.
+        :param num_samples: Number of sample draws of the minimal value.
+        :param grid_size: Size of random grid used to fit the gumbel distribution 
+            (recommend scaling with search space dimension).
+        """
+        self._search_space = search_space
+
+        if num_samples <= 0:
+            raise ValueError(
+                f"num_samples must be positive, got {num_samples}"
+            )
+        self._num_samples = num_samples
+
+        if grid_size <= 0:
+            raise ValueError(
+                f"num_samples must be positive, got {grid_size}"
+            )
+        self._grid_size = grid_size
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ModelInterface
+    ) -> AcquisitionFunction:
+        """
+        Need to sample possible min-values from our posterior. 
+        To do this we implement a Gumbel sampler.
+        We approximate Pr(y*^hat<y) by Gumbel(a,b) then sample from Gumbel.
+
+        :param dataset: Unused.
```

Comment:
the dataset is used below (which has me notice the docstrings for EI need correcting)
```suggestion
        :param datasets: The data from the observer.
```